federation/cmd/federation-controller-manager/app/controllermanager.go

```diff
@@ -236,3 +256,33 @@ func hasRequiredResources(serverResources []*metav1.APIResourceList, requiredRes
 	}
 	return true
 }
+
+func createFederationNamespace(clientset *federationclientset.Clientset, namespace string) error {
```

Comment:
This should be called "ensureFederationNamespace" because that's what it does.  If the namespace already exists, it does nothing, otherwise it tries to create it.